src/remote/server/server.cpp

```diff
@@ -406,6 +413,151 @@ class FailedLogins : private SortedObjectsArray<FailedLogin,
 	}
 };
 
+
+// For Trace
+
+class TraceFailedAuthImpl :
```

Comment:
Is it necessary to pollute server.cpp with such aux classes ? Could it be moved into separate file ?

Reply:
I see no reasons to move that class out of server.cpp. Or we should review a great lot of our sources - for example, TraceFailedConnection (cousin of TraceFailedAuthImpl) successfully lives in jrd.cpp.